transaction/src/action/ibc.rs

```diff
@@ -0,0 +1,77 @@
+use ibc::Height;
+use penumbra_crypto::{value, Address, Fr, Zero};
+use penumbra_proto::{ibc as pb, Protobuf};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(try_from = "pb::Ics20Withdrawal", into = "pb::Ics20Withdrawal")]
+pub struct ICS20Withdrawal {
+    // the chain ID of the destination chain for this ICS20 transfer
+    destination_chain_id: String,
+    // a transparent value consisting of an amount and an asset ID.
+    value: value::Value,
+    // the address on the destination chain to send the transfer to
+    destination_chain_address: String,
+    // a "sender" penumbra address to use to return funds from this withdrawal.
+    // this should be an ephemeral address
+    sender: Address,
+    // the height (on Penumbra) at which this transfer expires (and funds are sent
+    // back to the sender address?). NOTE: if funds are sent back to the sender,
+    // we MUST verify a nonexistence proof before accepting the timeout, to
+    // prevent relayer censorship attacks. The core IBC implementation does this
+    // in its handling of validation of timeouts.
+    timeout_height: Height,
+    // the timestamp at which this transfer expires.
+    timeout_time: u64,
+}
+
+impl ICS20Withdrawal {
+    pub fn value_commitment(&self) -> value::Commitment {
+        // we need to compute a value commitment to the transparent value of this withdrawal
+        let withdrawal_value = self.value.commit(Fr::zero());
+
+        // consume the withdrawal value. this should be negative because it consumes the value
+        // balance of the transaction. when we compute the binding verification key, we sum all of
+        // the value commitments. thus, there should be a spend that ouputs a positive value
+        // balance to this tx prior to this withdrawal action.
+        -withdrawal_value
```

Comment:
Yes, this is correct -- I'd tweak some of the wording in the comment though: it consumes *from* the transaction's value balance, rather than consuming the value balance itself.